CoreScriptsRoot/CoreScripts/MainBotChatScript2.lua

```diff
@@ -687,32 +692,64 @@ local getClosestDialogToPosition = guiService.GetClosestDialogToPosition
 game:GetService("RunService").Heartbeat:connect(function()
 	local closestDistance = math.huge
 	local closestDialog = nil
-	if usingGamepad == true then
+	
+	if (distanceOffsetFlag == false and usingGamepad == true) or (distanceOffsetFlag == true) then
 		if game:GetService("Players").LocalPlayer and game:GetService("Players").LocalPlayer.Character and game:GetService("Players").LocalPlayer.Character:FindFirstChild("HumanoidRootPart") then
 			local characterPosition = game:GetService("Players").LocalPlayer.Character:FindFirstChild("HumanoidRootPart").Position
 			closestDialog = getClosestDialogToPosition(guiService, characterPosition)
 		end
 	end
-
-	if closestDialog ~= lastClosestDialog then
-		if dialogMap[lastClosestDialog] then
-			dialogMap[lastClosestDialog].Background.ActivationButton.Visible = false
-		end
-		lastClosestDialog = closestDialog
-		contextActionService:UnbindCoreAction("StartDialogAction")
-		if closestDialog ~= nil then
-			contextActionService:BindCoreAction("StartDialogAction", function(actionName, userInputState, inputObject)
-				if userInputState == Enum.UserInputState.Begin then
-					if closestDialog and closestDialog.Parent then
-						startDialog(closestDialog)
-					end
-				end
-			end, false, Enum.KeyCode.ButtonX)
-			if dialogMap[closestDialog] then
-				dialogMap[closestDialog].Background.ActivationButton.Visible = true
+	
+	if distanceOffsetFlag then
+		if game:GetService("Players").LocalPlayer and 
+				game:GetService("Players").LocalPlayer.Character and
+				game:GetService("Players").LocalPlayer.Character:FindFirstChild("HumanoidRootPart") and
+				closestDialog ~= nil and
+				closestDialog.Parent ~= nil and
+				closestDialog.Parent:IsA("BasePart") and
+				currentConversationDialog == nil then
```

Comment:
Since this is a one time check, I'd actually prefer for it not to require any jumps to read. It's a unique check too - no other functionality needs this exact check.